src/neo/Trie/MPT/MPTNode.cs

```diff
@@ -0,0 +1,90 @@
+using Neo.Cryptography;
+using Neo.IO.Json;
+using System.IO;
+using System.Text;
+
+namespace Neo.Trie.MPT
+{
+    public abstract class MPTNode
+    {
+        private UInt256 hash;
+        public bool Dirty { get; private set; }
+        protected NodeType nType;
+
+        protected virtual UInt256 GenHash()
+        {
+            return new UInt256(Crypto.Hash256(this.Encode()));
+        }
+
+        public virtual UInt256 GetHash()
+        {
+            if (!Dirty && !(hash is null)) return hash;
+            hash = GenHash();
+            Dirty = false;
+            return hash;
+        }
+
+        public void SetDirty()
+        {
+            Dirty = true;
+        }
+
+        public MPTNode()
+        {
+            Dirty = true;
+        }
+
+        public byte[] Encode()
+        {
+            using (MemoryStream ms = new MemoryStream())
+            using (BinaryWriter writer = new BinaryWriter(ms, Encoding.UTF8))
```

Comment:
If we create two streams, or we pass `false` true `leaveOpen` in the constructor of BinaryWriter. Or we call twice `Dispose` in the `MemoryStream`

You have an example here:

```csharp
		var ms=new MemoryStream(new byte[1]);
		var st=new BinaryReader(ms,Encoding.ASCII);
		st.Dispose();
		Console.WriteLine($"{ms.ReadByte()}");
```

```
Unhandled exception. System.ObjectDisposedException: Cannot access a closed Stream.
   at System.IO.MemoryStream.ReadByte()
   at Program.Main()
```

Reply:
If you mean dispose twice, it's not a problem. I tried this:
```
            var ms = new MemoryStream(new byte[1]);
            var st = new BinaryReader(ms, Encoding.ASCII);
            st.Dispose();
            ms.Dispose();
```

Reply:
Not a problem, a loose of performance